### PR TITLE
Ensure Lightning test wheels are used in docker image

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [x] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** The docker build on PennyLane can fail tests due to the released version of Lightning being pulled in. As both repositories require synchronized changes, building a docker image from PennyLane master should also use the latest version of Lightning.

**Description of the Change:** Ensure the Lightning wheel is a pre-release from the latest master.

**Benefits:** Allows previously failing docker tests to pass.

**Possible Drawbacks:** Docker builds require the master version of PennyLane to be in sync.

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Docker builds
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (josh146) [inline .github/workflows/docker.yml:6]: @mlxd does it make sense to run this on all PRs? Since the tests are already being run on PRs.

Maybe it makes sense to only run this on the `master` branch?
```suggestion
  push:
    branches:
      - master
```

**Review by josh146 — CHANGES_REQUESTED** [review]
Looks good to me @mlxd! Requesting changes mainly because of the GH action trigger, I'm not sure if this is something we want to run on all PRs 🤔

File: docker/pennylane.dockerfile [hunk 2]
```diff
@@ -43,7 +43,8 @@ RUN git submodule update --init --recursive \
     && pip install wheel && pip install -r requirements.txt \
     && python3 setup.py install \
     && pip install pytest pytest-cov pytest-mock flaky \
-    && make test
+    && pip install -i https://test.pypi.org/simple/ pennylane-lightning --pre --upgrade \
```
Comment (albi3ro) [inline docker/pennylane.dockerfile:46]: Would this line have to be reverted each release cycle than put back in?

File: docker/qchem.dockerfile [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-From pennylane/base:latest
+FROM pennylane/base:latest
```
Comment (albi3ro) [inline docker/qchem.dockerfile:15]: why this change?

File: docker/qchem.dockerfile [hunk 4]
```diff
@@ -25,7 +25,8 @@ WORKDIR /opt/pennylane/qchem
 RUN git submodule update --init --recursive
 RUN pip install wheel && pip install openfermionpyscf && pip install -r requirements.txt \
     && python3 setup.py install \
-    && make test
+    && pip install -i https://test.pypi.org/simple/ pennylane-lightning --pre --upgrade \
+    && make test && make coverage
```
Comment (albi3ro) [inline docker/qchem.dockerfile:29]: Same here.  Will we need to revert this for each release?

**Review by albi3ro — COMMENTED** [review]
Just a question about whether this will need to become a task for our release cycle.

File: docker/pennylane.dockerfile (hunk 2 above)
Comment (mlxd) [reply docker/pennylane.dockerfile:46]: Since the docker image is built only against the master version of PennyLane, this should always pull in the latest master version of Lightning too.

File: docker/qchem.dockerfile (hunk 3 above)
Comment (mlxd) [reply docker/qchem.dockerfile:15]: It is good practice to have all docker syntax in BOLDFACE, and is used this way throughout the other dockerfiles.

File: docker/qchem.dockerfile (hunk 4 above)
Comment (mlxd) [reply docker/qchem.dockerfile:29]: As above.

**Review by josh146 — APPROVED** [review]
